pkg/policy/distillery/policy.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distillery
+
+import (
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/identity/cache"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+type cachedSelectorPolicy struct {
+	users  map[Endpoint]struct{}
```

Comment:
This map is not being protected by any mutex, it's confusing to understand that this should be protected by the mutex in `[1]`

Reply:
I'll add a comment, hopefully this makes it a bit clearer.